tests/arch/x86/cpu_scrubs_regs/src/main.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <syscall_handler.h>
+#include <ztest.h>
+#include "test_syscalls.h"
+
+#define DB_VAL 0xDEADBEEF
+
+void z_impl_test_cpu_write_reg(void)
+{
+	/* User thread CPU write registers system call for testing
+	*
+	* Verify the following
+	* - Write 0xDEADBEEF values during system call into registers
+	* - In main test we will read that registers to verify
+	* that all of them were scrubbed and do not contain any sensitive data
+	*/
+
+	/* Part below is made to test that kernel scrubs CPU registers
+	 * after returning from the system call
+	 */
+#if CONFIG_X86
+#ifndef CONFIG_X86_64
+	__asm__ volatile (
+		"movl $0xDEADBEEF, %eax;\n\t"
+		"movl $0xDEADBEEF, %ebx;\n\t"
+		"movl $0xDEADBEEF, %ecx;\n\t"
+		"movl $0xDEADBEEF, %edx;\n\t"
+		"movl $0xDEADBEEF, %edi;\n\t"
+		);
+#endif
+#endif
+#if CONFIG_X86
+#if CONFIG_X86_64
+	__asm__ volatile (
+		"movq $0xDEADBEEF, %rax;\n\t"
+		"movq $0xDEADBEEF, %rcx;\n\t"
+		"movq $0xDEADBEEF, %rdx;\n\t"
+		"movq $0xDEADBEEF, %rsi;\n\t"
+		"movq $0xDEADBEEF, %rdi;\n\t"
+		"movq $0xDEADBEEF, %r8;\n\t"
+		"movq $0xDEADBEEF, %r9;\n\t"
+		"movq $0xDEADBEEF, %r10;\n\t"
+		"movq $0xDEADBEEF, %r11;\n\t"
+		);
+#endif
+#endif
+}
```

Comment:
how about:
#if CONFIG_X86
#ifndef CONFIG_X86_64
#else
#endif //CONFIG_X86_64
#endif //CONFIG_X86

Reply:
@mengxianglinx I changed that in both places, in void z_impl_test_cpu_write_reg and in void test_syscall_cpu_scrubs_regs